### PR TITLE
Added tunnel/DHT info to debug panel

### PR DESCRIPTION
**Pull request by egbertbouman**

### CODE REVIEW

**Comment by egbertbouman** [comment]
retest this please

**Comment by qstokkink** [comment]
Can you chuck in a screenshot of this new functionality?

**Comment by egbertbouman** [comment]
@qstokkink Will do, but I'll have to update the fake-tribler-api first.

File: TriblerGUI/debug_window.py [hunk 1]
```diff
@@ -279,6 +293,52 @@ def on_ipv8_community_stats(self, data):
...
+
+    def on_relays_circuits(self, data):
+        self.add_items_to_tree(self.window().relays_tree_widget, data["relays"],
+                               ["circuit_from", "circuit_to", "is_rendezvous", "bytes_up", "bytes_down"])
+
+    def load_tunnel_exits_tab(self):
+        self.request_mgr = TriblerRequestManager()
+        self.request_mgr.perform_request("ipv8/tunnel/exits", self.on_tunnel_exits)
+
+    def on_tunnel_exits(self, data):
```
Comment (qstokkink) [inline TriblerGUI/debug_window.py]: data can be `None`

File: TriblerGUI/debug_window.py [hunk 2]
```diff
@@ -279,6 +293,52 @@ def on_ipv8_community_stats(self, data):
...
+
+    def on_tunnel_circuits(self, data):
+        self.add_items_to_tree(self.window().circuits_tree_widget, data["circuits"],
+                               ["circuit_id", "goal_hops", "actual_hops", "type", "state", "bytes_up", "bytes_down"])
+
+    def load_tunnel_relays_tab(self):
+        self.request_mgr = TriblerRequestManager()
+        self.request_mgr.perform_request("ipv8/tunnel/relays", self.on_relays_circuits)
+
+    def on_relays_circuits(self, data):
```
Comment (qstokkink) [inline TriblerGUI/debug_window.py]: `data` can be `None`

File: TriblerGUI/debug_window.py [hunk 3]
```diff
@@ -279,6 +293,52 @@ def on_ipv8_community_stats(self, data):
...
+            for index, key in enumerate(keys):
+                value = format_size(item[key]) if key in ["bytes_up", "bytes_down"] else str(item[key])
+                widget_item.setText(index, value)
+            tree.addTopLevelItem(widget_item)
+
+    def load_tunnel_circuits_tab(self):
+        self.request_mgr = TriblerRequestManager()
+        self.request_mgr.perform_request("ipv8/tunnel/circuits", self.on_tunnel_circuits)
+
+    def on_tunnel_circuits(self, data):
```
Comment (qstokkink) [inline TriblerGUI/debug_window.py]: `data` can be `None`

**Comment by qstokkink** [comment]
retest this please. Both the Linux and Windows failure are well-known test suite errors.

**Comment by egbertbouman** [comment]
retest this please

**Comment by synctext** [comment]
Perfect, ready for RC2 hopefully.

**Comment by devos50** [comment]
@synctext yes, this will be included in RC2.

**Pull request merged**